### DISCUSSION TITLE
이벤트 위임 (Event Delegation)

### DISCUSSION THREAD

**Discussion by greenblues1190** [woowacourse-study/2022-woowahan-bansanghwe#7] [JavaScript]
이벤트 위임에 대해 설명해 주세요.

**Comment by rladpwl0512** [comment]
**질문 리스트** 

- **이벤트 캡처링은 어떤 객체부터 시작되나요?** 
`window` 객체부터 시작 
`window` -> `document` -> ... -> `target` 

- **`focus`, `blur`, `mouseenter`, mouseleave`는` 버블링이 되지 않는데, focus를 버블링 시키고 싶을 때는 어떻게 해야할까요?**
-> 이벤트를 상위 요소에서 캐치해야할경우, 대체할 수 있는 이벤트가 존재한다.  **(deep dive 782p 참고)** 
![IMG_90A3965AE662-1](https://user-images.githubusercontent.com/52344833/165015184-8cc54904-3493-4b0f-b017-5806e4c9361b.jpeg)

- **🤔이벤트 캡처링은 어디에서 사용할 수 있을까요?** 
(우디-브라우저에 따라 다르다. / 병민-더 뭔가 있을 것 같은데... ui / 마르코 - 이벤트 핸들러가 등록되는 순서) 
어렵다! 정리 필요

- **이벤트 버블링은 어디에서 사용할 수 있을까요?**
이벤트 위임 

- **이벤트 위임의 장단점?**
연산이 많이 든다 
단점을 개선하기 위해, 최대한 가까운 부모요소에 이벤트를 걸어주자

**Comment by wonsss** [comment]
[추가 질문]
- 이벤트 캡처링은 어디서부터 시작하나요? 
- 거의 모든 이벤트가 버블링되고, 버블링되지 않는 이벤트도 있다고 하는데, 왜 어떤 이벤트는 버블링되지 않나요? 
- 캡처링은 언제 쓰나요?
